torch/testing/_internal/common_methods_invocations.py

```diff
@@ -950,19 +950,21 @@ def method_tests():
         ('prod', (), NO_ARGS, 'scalar'),
         ('prod', (), (0,), 'scalar_dim', (), [0]),
         ('prod', (), (0, True,), 'scalar_keepdim_dim', (), [0]),
-        ('prod', prod_zeros(S, [0, 1]), NO_ARGS, 'zerodims2'),
-        ('prod', prod_zeros(S, [0, 2]), NO_ARGS, 'zerodims1'),
-        ('prod', prod_zeros(S, [1, 2]), NO_ARGS, 'zerodims0'),
-        ('prod', prod_zeros(S, [0, 1]), (1,), 'zeros_dims2', (), [0]),
-        ('prod', prod_zeros(S, [0, 2]), (1,), 'zeros_dims1', (), [0]),
-        ('prod', prod_zeros(S, [1, 2]), (1,), 'zeros_dims0', (), [0]),
-        ('prod', prod_zeros(S, [0, 1]), (1, True), 'keepdim_zeros_dims2', (), [0]),
-        ('prod', prod_zeros(S, [0, 2]), (1, True), 'keepdim_zeros_dims1', (), [0]),
-        ('prod', prod_zeros(S, [1, 2]), (1, True), 'keepdim_zeros_dims0', (), [0]),
-        ('prod', prod_single_zero(S), NO_ARGS, 'single_zero'),
-        ('prod', (torch.tensor(0., requires_grad=True)), NO_ARGS, 'scalar_zero'),
-        ('prod', (torch.tensor(0., requires_grad=True)), (0,), 'scalar_dim_zero', (), [0]),
-        ('prod', (torch.tensor(0., requires_grad=True)), (0, True,), 'scalar_keepdim_dim_zero', (), [0]),
+        ('prod', lambda dtype, device: prod_zeros(S, [0, 1], dtype), NO_ARGS, 'zerodims2'),
+        ('prod', lambda dtype, device: prod_zeros(S, [0, 2], dtype), NO_ARGS, 'zerodims1'),
+        ('prod', lambda dtype, device: prod_zeros(S, [1, 2], dtype), NO_ARGS, 'zerodims0'),
+        # The tests below rely on index_add, which is not yet implemented on CUDA for complex types
+        # TODO(@nikitaved): enable these tests later
+        ('prod', lambda dtype, device: prod_zeros(S, [0, 1], dtype), (1,), 'zeros_dims2', (), [0]),
+        #('prod', lambda dtype, device: prod_zeros(S, [0, 2], dtype), (1,), 'zeros_dims1', (), [0]),
```

Comment:
You actually disable the non-complex tests as well here? That does not sound good

Reply:
It is still WIP, I need `index_add` complex dispatch on CUDA for this to work.